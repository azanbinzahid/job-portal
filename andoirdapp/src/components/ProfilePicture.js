import React from 'react';
import {connect} from 'react-redux';
import {Image, Form, Button, Container} from 'react-bootstrap';
import {useFormik} from 'formik';
import {uploadImage} from 'redux/actions';

const ProfilePicture = (props) => {
  const {handleSubmit, setFieldValue} = useFormik({
    initialValues: {image: ''},
    onSubmit(values) {
      props.uploadImage(values);
    },
  });

  return (
    <Container align="center" className="pb-5">
      <Image src={process.env.REACT_APP_BASE_URL + props.profileImage} fluid />

      <Form inline onSubmit={handleSubmit}>
        <Form.Group controlId="formBasicDate">
          <Form.File
            name="image"
            onChange={(event) => {
              setFieldValue('image', event.currentTarget.files[0]);
            }}
          />
        </Form.Group>

        <Button variant="primary" type="submit">
          Update Image
        </Button>
      </Form>
    </Container>
  );
};

const mapStateToProps = (state) => ({
  profileImage: state.userReducer.user.profile.image,
});

const mapDispatchToProps = (dispatch) => {
  return {
    uploadImage: (userInfo) => dispatch(uploadImage(userInfo)),
  };
};

export default connect(mapStateToProps, mapDispatchToProps)(ProfilePicture);
