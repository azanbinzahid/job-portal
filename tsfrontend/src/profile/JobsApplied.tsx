import React, { FC } from "react";
import { ListGroup, ListGroupItem } from "react-bootstrap";
import { LinkContainer } from "react-router-bootstrap";
import { Job } from "redux/types";

type Props = {
  jobsApplied: Job[];
};

const JobsApplied: FC<Props> = ({ jobsApplied }) => {
  return (
    <ListGroup>
      <h3> Jobs Applied </h3>
      {jobsApplied.map((job, index) => (
        <LinkContainer key={index} to={`/jobs/${job.id}`}>
          <ListGroupItem action key={index}>
            {job.title}
          </ListGroupItem>
        </LinkContainer>
      ))}
    </ListGroup>
  );
};

export default JobsApplied;
