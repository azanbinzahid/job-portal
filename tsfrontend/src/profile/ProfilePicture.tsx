import React, { FC } from "react";
import { connect } from "react-redux";
import { Image, Form, Button, Container } from "react-bootstrap";
import { useFormik } from "formik";
import { uploadImage } from "redux/actions";
import { RootState } from "redux/reducers";

type Props = {
  profileImage: string;
  uploadImage: (values: { image: string }) => void;
};
const ProfilePicture: FC<Props> = (props) => {
  const { handleSubmit, setFieldValue } = useFormik({
    initialValues: { image: "" },
    onSubmit(values) {
      props.uploadImage(values);
    },
  });

  return (
    <Container<React.ElementType> align="center" className="pb-5">
      <Image src={process.env.REACT_APP_BASE_URL + props.profileImage} fluid />
      <Form
        inline
        onSubmit={(e: React.FormEvent<HTMLFormElement>) => handleSubmit(e)}
      >
        <Form.File<React.ElementType>
          name="image"
          onChange={(event: any) => {
            setFieldValue("image", event.currentTarget.files[0]);
          }}
        />

        <Button variant="primary" type="submit">
          Update Image
        </Button>
      </Form>
    </Container>
  );
};

const mapStateToProps = (state: RootState) => ({
  profileImage: state.userReducer.user.profile.image,
});

const mapDispatchToProps = {
  uploadImage,
};

export default connect(mapStateToProps, mapDispatchToProps)(ProfilePicture);
