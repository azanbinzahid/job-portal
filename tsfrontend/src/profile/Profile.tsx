import React, { FC } from "react";
import { connect } from "react-redux";
import { Container, Row, Col } from "react-bootstrap";
import EditProfileForm from "profile/EditProfileForm";
import ProfilePicture from "profile/ProfilePicture";
import JobsApplied from "profile/JobsApplied";
import { RootState } from "redux/reducers";
import { Job } from "redux/types";
import SectionHeading from "app/common/SectionHeading";

type Props = {
  firstName: string;
  lastName: string;
  profileImage: string;
  jobsApplied: Job[];
};
export const Profile: FC<Props> = (props) => {
  return (
    <Container>
      <SectionHeading
        title={props.firstName + " " + props.lastName}
        subtitle="Profile Page"
      />
      <Row>
        <Col md="7">
          <EditProfileForm />
        </Col>
        <Col md="5">
          <ProfilePicture />
          <JobsApplied jobsApplied={props.jobsApplied} />
        </Col>
      </Row>
    </Container>
  );
};

const mapStateToProps = (state: RootState) => ({
  firstName: state.userReducer.user.firstName,
  lastName: state.userReducer.user.lastName,
  profileImage: state.userReducer.user.profile.image,
  jobsApplied: state.userReducer.user.jobsApplied,
});

export default connect(mapStateToProps)(Profile);
