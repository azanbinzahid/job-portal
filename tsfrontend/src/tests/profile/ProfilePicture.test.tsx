import React from "react";
import { shallow } from "enzyme";
import { ProfilePicture } from "profile/ProfilePicture";

test("renders the component", () => {
  const component = shallow(
    <ProfilePicture profileImage="" uploadImage={jest.fn()} />
  );
  expect(component).toMatchSnapshot();
});
