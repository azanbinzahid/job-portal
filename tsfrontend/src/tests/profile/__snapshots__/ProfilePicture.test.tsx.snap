// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the component 1`] = `
<Container
  align="center"
  className="pb-5"
  fluid={false}
>
  <Image
    fluid={true}
    rounded={false}
    roundedCircle={false}
    src="http://localhost:8000"
    thumbnail={false}
  />
  <Form
    inline={true}
    onSubmit={[Function]}
  >
    <FormFile
      name="image"
      onChange={[Function]}
    />
    <Button
      active={false}
      disabled={false}
      type="submit"
      variant="primary"
    >
      Update Image
    </Button>
  </Form>
</Container>
`;
